discord/app_commands/commands.py

```diff
@@ -934,8 +934,8 @@ def __init__(
         self._children: Dict[str, Union[Command, Group]] = {}
 
         for child in self.__discord_app_commands_group_children__:
-            child.parent = self
             child = child._copy_with_binding(self) if not cls.__discord_app_commands_skip_init_binding__ else child
+            child.parent = self
```

Comment:
I originally had these lines like this but I swapped them because the child would not copy the parent if it's done after the `_copy_with_binding` call. What's the reason for swapping it back?

In my mental model with a setup like `/foo`, `/foo bar`, `/foo bar baz`

Old order:

- `/foo` gets copied. 
- Parent for `bar` is set to `/foo`. 
- `bar` is copied (parent properly assigned)
- Bug: Children are copied (but parent is not properly assigned)

New order:

- `/foo` gets copied
- `bar` parent is the *old* `/foo`
- `bar` is copied
- children are copied and their parent is set to the copy

Unless I'm missing something I think the first layer for `/foo bar` points to the old `/foo`?

Reply:
> What's the reason for swapping it back?

In the case where a copy is made on line 937, setting the parent before the copy is made is updating the class child's parent reference, so `MyGroup.__discord_app_commands_group_children__[0].parent` becomes the latest instance of `MyGroup`.

I'm going to put together a test to make sure all of the scenarios are properly covered by this new code because I think you're right that this new order misses something.